### PR TITLE
[Transforms] Performance improvements in the comment emitter.

### PR DESCRIPTION
**Pull request by rbuckton**
Simplifies the comment emitter and reduces allocations for Comment Ranges. Also allows the `runtests-file` jake task to run in parallel.

### CODE REVIEW

**Comment by rbuckton** [comment]
CC: @vladima, @yuit, @sandersn, @mhegazy

File: src/compiler/comments.ts [hunk 1]
```diff
@@ -269,6 +287,18 @@ namespace ts {
...
+        }
+
+        function disableCommentsAndEmit(node: Node, emitCallback: (node: Node) => void): void {
+            if (disabled) {
+                emitCallback(node);
+            }
+            else {
+                disabled = true;
+                emitCallback(node);
+                disabled = false;
```
Comment (vladima) [inline src/compiler/comments.ts]: maybe it should save previous value of `disabled` and restore it later instead of unconditionally set it to `false`?

Comment (rbuckton) [inline src/compiler/comments.ts]: There is no need, the first branch handles that case.

Comment (vladima) [inline src/compiler/comments.ts]: got it

File: src/compiler/scanner.ts [hunk 2]
```diff
@@ -594,10 +594,15 @@ namespace ts {
      * and the next token are returned.
      * If true, comments occurring between the given position and the next line break are returned.
```
Comment (yuit) [inline src/compiler/scanner.ts]: should these comment get updated?

File: src/compiler/scanner.ts [hunk 3]
```diff
@@ -647,37 +654,76 @@ namespace ts {
...
-        return result;
+        if (hasPendingCommentRange) {
+            accumulator = cb(pendingPos, pendingEnd, pendingKind, pendingHasTrailingNewLine, state, accumulator);
+        }
+
+        return accumulator;
+    }
+
+    export function forEachLeadingCommentRange<T, U>(text: string, pos: number, cb: (pos: number, end: number, kind: SyntaxKind, hasTrailingNewLine: boolean, state: T) => U, state?: T) {
+        return iterateCommentRanges(/*reduce*/ false, text, pos, /*trailing*/ false, cb, state);
```
Comment (yuit) [inline src/compiler/scanner.ts]: why u return anything here? from searching around, it doesn't seem to be consumed by anything

Comment (rbuckton) [inline src/compiler/scanner.ts]: I had investigated using this also for detached comments and was using the early-exit capability but decided against that approach. I left it in for consistency with `forEach` in core.

**Comment by yuit** [comment]
:+1:

**Pull request merged**